### PR TITLE
retire: Use Log4j 2.x

### PR DESCRIPTION
**Pull request by kingthorin**
- Update CHANGELOG, and change code to use updated logging infrastructure.

Part of: zaproxy/zaproxy#6376

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireScanRule.java [hunk 1]
```diff
@@ -68,23 +69,15 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
...
-                                    + " and version:"
-                                    + result.getVersion()
-                                    + ", more info at:"
-                                    + result.getInfo());
-                }
+                LOGGER.debug(
+                        "\tVulnerabilities found in record {} with URL: {}, name: {} and version: {}, more info at: {}",
+                        id,
+                        uri,
+                        ((result.getFilename() == null) ? "" : result.getFilename()),
```
Comment (thc202) [inline addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireScanRule.java:78]: The parentheses could be removed now.

**Comment by thc202** [comment]
The commit message says that the build file and changelog were updated but they were not changed.

**Comment by kingthorin** [comment]
Oh good point, that was bad re-use of a commit message on my part. I'll tidy it up.

**Comment by kingthorin** [comment]
Tweaked.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**